components/org.wso2.carbon.crypto.impl/src/main/java/org/wso2/carbon/crypto/impl/DefaultCryptoService.java

```diff
@@ -1006,4 +1065,50 @@ private void failIfHybridDecryptOperationInputsAreInvalid(HybridEncryptionOutput
             throw new CryptoException(errorMessage);
         }
     }
+
+    private void failIfInternalCryptoInputsAreNotValid(byte[] data, String algorithm, String internalCryptoProviderType,
+                                                       String operation) throws CryptoException {
+
+        if (data == null) {
+            throw new CryptoException(String.format("Content provided for the %s operation can't be null", operation));
+        }
```

Comment:
Is this correct. null is checked for data but log talks about operation

Reply:
Yes it is correct. data is a byte[]. We are saying the data provided for operation "Internal Encryption" cannot be null